### PR TITLE
Eliminate "init" command in favor of "start" only with -bootstrap flag(s...

### LINKED ISSUE #662: Cockroach Docker walkthrough is broken
* we need to pass `-init_and_start` (btw, that looks ugly with the underscores, and redundant since the flag is really only used with `init`. Why do we have both `start` and `init` anyways? Could just have `start` and merge the current behavior into `--bootstrap` and `--bootstrap-only` flags.

* `./cockroach put a 2` is now broken because it's trying to POST to the empty host. It should probably default to localhost (or to whatever the RPC server determines its hostname is). We had issues with stuff like this before.

We should find a way to make sure that the client examples we give work at all times, i.e. by somehow adding them to the test suite.

### PR DESCRIPTION
**Pull request by spencerkimball**
...).

Removed -init_and_start and added -bootstrap and -bootstrap-only flags.
Also cleaned up error logging in Store.Bootstrap to make more sense.

Changed multi-word flags to using hyphens instead of underscores.

Updated docs and local-cluster.sh script.

Fixes #662

### CODE REVIEW

**Comment by bdarnell** [comment]
I'd actually prefer to get rid of the init-and-start combination entirely, and require the use of two separate commands. A combined init-and-start interacts poorly with process managers (e.g. [docker-compose](https://github.com/cockroachdb/cockroach-python/blob/68bbdc94e53728f79adcf68780b3bf4433d482d7/docker-compose.yml), since when the "init" command exits it must be resumed as "start". I think it's better to keep things separate and explicit, with an "init" command that returns immediately and is run once, and then the long-running processes are always started the same way, with "start".

File: storage/store.go [hunk 1]
```diff
@@ -581,9 +581,17 @@ func (s *Store) Bootstrap(ident proto.StoreIdent, stopper *util.Stopper) error {
 	s.Ident = ident
 	kvs, err := engine.Scan(s.engine, proto.EncodedKey(engine.KeyMin), proto.EncodedKey(engine.KeyMax), 1)
 	if err != nil {
-		return util.Errorf("unable to scan engine to verify empty: %s", err)
+		return util.Errorf("store %s: unable to access: %s", s.engine, err)
 	} else if len(kvs) > 0 {
-		return util.Errorf("non-empty engine %s (first key: %q)", s.engine, kvs[0].Key)
+		// See if this is an already-bootrapped store.
```
Comment (tschottdorf) [inline storage/store.go]: s/trap/strap/

**Comment by tschottdorf** [comment]
LGTM

**Comment by spencerkimball** [comment]
PTAL

File: server/cli/start.go [hunk 2]
```diff
@@ -169,6 +160,7 @@ func runStart(cmd *commander.Command, args []string) {
 	select {
 	case <-stopper.ShouldStop():
 		stopper.SetStopped()
+		close(gracefulShutdownCh)
```
Comment (MBerhault) [inline server/cli/start.go]: We can't close this yet. You need to wait until all the Stopper.closers have been called.
Two options (at least):
- have stopper.Stop() be callable multiple times (you would need the mutex to be held on the closer runners and probably a noop check at the beginning)
- have a new channel to signal that Stop() has finished (Stopper.stopper is not enough, we need the closers too)

Comment (tschottdorf) [inline server/cli/start.go]: @spencerkimball in my commits to #636 I had the first option implemented and working -- basically just holding the lock in `Stop()` and doing nothing if the thing's state indicated it was already done.

File: README.md [hunk 3]
```diff
@@ -144,8 +146,8 @@ Once you've built your image, you may want to run the tests:
 Assuming you've built `cockroachdb/cockroach`, let's run a simple Cockroach node in the background:
 
 ```bash
-$ docker run -p 8080:8080 -d cockroachdb/cockroach init \
-    -stores ssd="$(mktemp -d /tmp/dbXXX)"
+$ docker run -p 8080:8080 -d -v /cr-data cockroachdb/cockroach init /cr-data
```
Comment (tschottdorf) [inline README.md]: you don't want to daemonize the first one. Besides, don't you do an init-and-start in the next line? Seems like you don't need to run the first command and pass the volume if you do that.

Comment (tschottdorf) [inline README.md]: (and even if you did, I thought we needed to get the CID and do `--volumes-from`)

**Comment by tschottdorf** [comment]
Can we make sure that we clean up the docs to avoid things like #672 with this PR as well?

**Comment by tschottdorf** [comment]
notably (and this has been broken more than once):

[ ] if `-gossip` is not specified, use the right thing for the local node
[ ] make sure that the walkthroughs work.

I think it's a good idea to not have to pass all sorts of parameters to just start a single node for playing around.

**Comment by tschottdorf** [comment]
also #662

File: server/cli/start.go (hunk 2 above)
Comment (spencerkimball) [inline server/cli/start.go]: I've added a new channel to the stopper to accomplish the goal.

File: README.md (hunk 3 above)
Comment (spencerkimball) [inline README.md]: Sorry, this bit wasn't updated to reflect recent changes. Updated now.

**Pull request merged**